src/energyplus/Test/PythonPluginInstance_GTest.cpp

```diff
@@ -0,0 +1,130 @@
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+#include "EnergyPlusFixture.hpp"
+
+#include "../ForwardTranslator.hpp"
+#include "../ReverseTranslator.hpp"
+
+#include "../../model/Model.hpp"
+#include "../../model/ExternalFile.hpp"
+#include "../../model/ExternalFile_Impl.hpp"
+#include "../../model/PythonPluginInstance.hpp"
+#include "../../model/PythonPluginInstance_Impl.hpp"
+
+#include "../../utilities/idf/Workspace.hpp"
+#include "../../utilities/idf/IdfObject.hpp"
+#include "../../utilities/idf/WorkspaceObject.hpp"
+#include "../../utilities/idf/IdfExtensibleGroup.hpp"
+
+#include "../../utilities/core/PathHelpers.hpp"
+
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/PythonPlugin_Instance_FieldEnums.hxx>
+#include <utilities/idd/PythonPlugin_SearchPaths_FieldEnums.hxx>
+
+#include <resources.hxx>
+
+#include <sstream>
+
+using namespace openstudio::energyplus;
+using namespace openstudio::model;
+using namespace openstudio;
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_PythonPluginInstance) {
+  Model model;
+
+  openstudio::path p = resourcesPath() / toPath("model/PythonPluginThermochromicWindow.py");
+  ASSERT_TRUE(exists(p));
+
+  // This is to avoid issues where if you change the original OpenStudio/resources/model/PythonPluginThermochromicWindow.py it will throw in
+  // getExternalFile because the file exists and it doesn't have the same content
+  path expectedDestDir;
+  std::vector<path> absoluteFilePaths = model.workflowJSON().absoluteFilePaths();
+  if (absoluteFilePaths.empty()) {
+    expectedDestDir = model.workflowJSON().absoluteRootDir();
+  } else {
+    expectedDestDir = absoluteFilePaths[0];
+  }
+
+  if (exists(expectedDestDir)) {
+    removeDirectory(expectedDestDir);
+  }
+  ASSERT_FALSE(exists(expectedDestDir));
+
+  boost::optional<ExternalFile> externalfile = ExternalFile::getExternalFile(model, openstudio::toString(p));
+  ASSERT_TRUE(externalfile) << "Path doesn't exist: '" << p << "'";
+
+  PythonPluginInstance pythonPluginInstance1(*externalfile, "ZN_1_wall_south_Window_1_Control");
+  PythonPluginInstance pythonPluginInstance2(*externalfile, "ZN_1_wall_south_Window_1_Control");
+  pythonPluginInstance2.setRunDuringWarmupDays(true);
+
+  ForwardTranslator ft;
+  Workspace workspace = ft.translateModel(model);
+
+  std::string errors;
+  for (const auto& error : ft.errors()) {
+    errors += error.logMessage() + "\n";
+  }
+  EXPECT_EQ(0, ft.errors().size()) << errors;
+
+  std::string warnings;
+  for (const auto& warning : ft.warnings()) {
+    warnings += warning.logMessage() + "\n";
+  }
+  EXPECT_EQ(0, ft.warnings().size()) << warnings;
```

Comment:
What happened to our test for not translating objects referencing files without a valid class name?

Reply:
Since I now disallow setting a bad value at model time it's harder to do, but you're right we should re-add that.

Reply:
DOne in 20150bb9d6708333917a2616e9cc1607b43b0ae0